### DISCUSSION TITLE
The easiest way to replace $...$ to \(...\)

### DISCUSSION THREAD

**Discussion by sleepymalc** [James-Yu/LaTeX-Workshop#3418] [Q&A] [answered]
I have been searching around for this seemingly basic feature, but I found nothing. Does anyone know how to do this? I'm imagining that there is some extension that can just do this on save, which will save **tons** of time. The closest thing I found is probably [this](https://github.com/James-Yu/LaTeX-Workshop/wiki/Environments#changing-between--and-beginend), which is by no means efficient.

**Comment by James-Yu** [comment, marked as answer]
You can use this regex in vscode to search for all $$ pairs: `(?<!\\)\$([^\r]*?)(?<!\\)\$`. Then `\($1\)` or something else can replace them with `\(` `\)`.

**Reply by sleepymalc** [reply]
Ahh I didn't know VS Code replace support syntax like `$1`. This helps!!! Though the regrex you gave will also detect `$$` in `$$ inline math $$`, just to point it out...

**Reply by James-Yu** [reply]
Just replace the star with plus + to avoid $$
